vnext/ReactUWP/Views/TextInputViewManager.cpp

```diff
@@ -232,102 +293,159 @@ void TextInputShadowNode::createView() {
-          textBox.Select(selection.start, selection.end - selection.start);
-      }
-    } else if (propertyName == "selectionColor") {
-      if (IsValidColorValue(propertyValue))
-        textBox.SelectionHighlightColor(SolidColorBrushFrom(propertyValue));
-      else if (propertyValue.isNull())
-        textBox.ClearValue(winrt::TextBox::SelectionHighlightColorProperty());
     } else if (propertyName == "selectTextOnFocus") {
       if (propertyValue.isBool())
         m_shouldSelectTextOnFocus = propertyValue.asBool();
-    } else if (propertyName == "spellCheck") {
-      if (propertyValue.isBool())
-        textBox.IsSpellCheckEnabled(propertyValue.asBool());
-      else if (propertyValue.isNull())
-        textBox.ClearValue(winrt::TextBox::IsSpellCheckEnabledProperty());
-    } else if (propertyName == "text") {
-      if (m_mostRecentEventCount == m_nativeEventCount) {
-        if (propertyValue.isString()) {
-          auto oldValue = textBox.Text();
-          auto newValue = asHstring(propertyValue);
-          if (oldValue != newValue) {
-            textBox.Text(newValue);
-          }
-        } else if (propertyValue.isNull())
-          textBox.ClearValue(winrt::TextBox::TextProperty());
-      }
     } else if (propertyName == "mostRecentEventCount") {
       if (propertyValue.isNumber()) {
         m_mostRecentEventCount = static_cast<uint32_t>(propertyValue.asInt());
       }
+    } else if (propertyName == "secureTextEntry") {
+      if (propertyValue.isBool()) {
+        if (propertyValue.asBool()) {
+          if (m_isTextBox) {
+            winrt::PasswordBox pwBox;
+            ReparentView(pwBox);
+            m_isTextBox = false;
+            RegisterEvents();
+            control = pwBox.as<winrt::Control>();
+          }
+        } else {
+          if (!m_isTextBox) {
+            winrt::TextBox textBox;
+            ReparentView(textBox);
+            m_isTextBox = true;
+            RegisterEvents();
+            control = textBox.as<winrt::Control>();
+          }
+        }
+      }
+    } else {
+      if (m_isTextBox) { // Applicable properties for TextBox
+        auto textBox = control.as<winrt::TextBox>();
+        if (TryUpdateTextAlignment(textBox, propertyName, propertyValue)) {
+          continue;
+        } else if (propertyName == "multiline") {
+          if (propertyValue.isBool())
+            textBox.TextWrapping(
+                propertyValue.asBool() ? winrt::TextWrapping::Wrap
+                                       : winrt::TextWrapping::NoWrap);
+          else if (propertyValue.isNull())
+            textBox.ClearValue(winrt::TextBox::TextWrappingProperty());
+        } else if (propertyName == "editable") {
+          if (propertyValue.isBool())
+            textBox.IsReadOnly(!propertyValue.asBool());
+          else if (propertyValue.isNull())
+            textBox.ClearValue(winrt::TextBox::IsReadOnlyProperty());
+        } else if (propertyName == "maxLength") {
```

Comment:
Is it possible to put the common property keys (like maxLength) into one clause like before, and just change which target property is used? Something like:

```
else if (propertyName == "maxLength") {
  if (propertyValue.isBool()) {
    control.SetValue(m_isTextBox ? winrt::TextBox::MaxLengthProperty() : winrt::PasswordBox::MaxLengthProperty(), static_cast<int32_t>(propertyValue.asDouble()));
  } else if (propertyValue.IsNull) {
    control.ClearValue(m_isTextBox ? winrt::TextBox::MaxLengthProperty() : winrt::PasswordBox::MaxLengthProperty());
  }
}
```